tests/lib/test_config.py

```diff
@@ -234,3 +234,55 @@ def test_resolve_variables_deep_with_invalid_reference(self, _io_mock):
                     ],
                 },
             )
+
+    def test_get_previous_step_with_previous_step(self, _io_mock):
+        """Should answer with the previous step."""
+        previous_step = config.get_previous_step({"workflow": ["step1", "step2", "step3"]}, "step2")
+        self.assertEqual(previous_step, "step1")
+
+    def test_get_previous_step_without_previous_step(self, _io_mock):
+        """Should answer with None as the previous step does not exist."""
+        previous_step = config.get_previous_step({"workflow": ["step1", "step2", "step3"]}, "step1")
+        self.assertIsNone(previous_step)
+
+    def test_get_previous_step_raises_error_with_incorrect_config(self, _io_mock):
+        """Should raise an error if config is malformed."""
+        with self.assertRaises(KeyError):
+            config.get_previous_step({}, "step1")
+
+    @patch("nestor_api.lib.config.os", autospec=True)
+    def test_list_apps_config(self, os_mock, io_mock):
+        """Should return an dictionary of apps config."""
+        os_mock.path.isdir.return_value = True
+        os_mock.listdir.return_value = ["path/to/app-1", "path/to/app-2", "path/to/app-3"]
+        os_mock.path.isfile.return_value = True
+
+        def yaml_side_effect(arg):
+            # pylint: disable=no-else-return
+            if arg == "path/to/app-1":
+                return {"name": "app-1", "config_key": "value for app-1"}
+            elif arg == "path/to/app-2":
+                return {"name": "app-2", "config_key": "value for app-2"}
+            elif arg == "path/to/app-3":
+                return {"name": "app-3", "config_key": "value for app-3"}
+            return None
+
+        io_mock.from_yaml.side_effect = yaml_side_effect
```

Comment:
```suggestion
        io_mock.from_yaml.side_effect = [
          {"name": "app-1", "config_key": "value for app-1"},
          {"name": "app-2", "config_key": "value for app-2"},
          {"name": "app-3", "config_key": "value for app-3"},
        ]
```

You can also do it this way (but your solution works fine, just FYI, also you check for the params that are given, this solution does not)